src/clj/puppetlabs/services/master/master_service.clj

```diff
@@ -19,8 +19,8 @@
          localcacert (get-in config [:puppet-server :localcacert])
          settings    (ca/config->master-settings config)]
 
-     (retrieve-ca-cert! localcacert)
      (initialize-master-ssl! settings certname)
+     (retrieve-ca-cert! localcacert)
```

Comment:
Yeah, @dankreek, I think you're right that it would have to go in this order now if `localcacert` is added to the `required-master-files`.  In the reverse order, the `cacert` would be copied to `localcacert` by the `retrieve-ca-cert!` function before `initialize-master-ssl!` is called.  `initialize-master-ssl!` would then throw a partial state error because the `localcacert` file already exists but the `hostprivkey` and `hostcert` do not.

It hadn't dawned on me to try to add `localcacert` to `required-master-files` as part of this PR but I suppose that makes sense.  I don't see any problems with reversing the order since `initialize-master-ssl!` shouldn't need the `localcacert` to exist.